src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -1286,5 +1283,7 @@ private void CheckValidInstance()
                 throw new InvalidOperationException();
             }
         }
+
+        private string DebuggerDisplay => $"TokenType = {TokenType} : \"{ToString()}\"";
```

Comment:
Should this be `ValueType` instead?

Reply:
Also, should we show ArrayLength if ValueType == array?

Reply:
And just like JsonProperty, should be special case `JsonElement element = default;` where type is none/undefined?

Reply:
> Should this be ValueType instead?

I thought TokenType was more interesting, but I'll look. @bartonjs?

> Also, should we show ArrayLength if ValueType == array?

The full source text comes from ToString, which seemed to capture more than any other thing we could add.

> And just like JsonProperty, should be special case JsonElement element = default; where type is none/undefined?

There wasn't a throw like the other one. I have tests for the uninitialized state to catch these.

Reply:
TokenType is an internal property Type (JsonValueType) is what a user would see.

Reply:
(And since rereading my comment doesn't feel clear to me: The debugger display should use Type, not TokenType)